pkg/resource/user_group/custom_update_api.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package user_group
+
+import (
+	"context"
+	"errors"
+	svcapitypes "github.com/aws-controllers-k8s/elasticache-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svcsdk "github.com/aws/aws-sdk-go/service/elasticache"
+)
+
+// Implements custom logic for UpdateUserGroup
+func (rm *resourceManager) customUpdateUserGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+
+	// Check for the user group status
+	if latest.ko.Status.Status == nil || *latest.ko.Status.Status != "active" {
+		return nil, requeue.NeededAfter(
+			errors.New("user group can not be modified, it is not in 'active' state"),
+			requeue.DefaultRequeueAfterDuration)
+	}
+
+	for _, diff := range delta.Differences {
+		if diff.Path.Contains("Spec.UserIDs") {
```

Comment:
I may have seen this comment elsewhere but I think this can be replaced by `delta.DifferentAt("Spec.UserIDs")`?

Reply:
Actually I think I was incorrect above. I think neither of the approaches are currently functional since `Path.Contains` loops through the parts and looks for an exact match with the subject string. The issue is that one part is "Spec" and one part is "UserIDs" but there is no part named "Spec.UserIDs".

The e2e test may have passed since there is no assertion that the new user is actually contained in the user group.

Reply:
https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/compare/delta.go#L28-L35

The current implementation of `delta.DifferentAt` is potentially problematic IMO. Changing the above code to `delta.DifferentAt("UserIDs")` may make it function, but the risk is that there could some other field (e.g. in `status`, `metadata`) or sub-fields also named `UserIDs` that would fall under this call, not just the field of interest in `spec`.

One possible solution would be to split the supplied subject string and only return true if the resulting slice matches `diff.Path.parts` exactly (in length, content, and order). This seems closer to the intention conveyed in the comment above `DifferentAt`. Relying on `Path.Contains` may be too imprecise.

@jaypipes could you give your input on this issue?

Reply:
see Amine's comment above (Issue 808)

Reply:
> @jaypipes could you give your input on this issue?

@echen-98 no need for me to comment on this now that https://github.com/aws-controllers-k8s/runtime/pull/20 is merged, right?

Reply:
@jaypipes correct